gpiozero/output_devices.py

```diff
@@ -300,7 +299,10 @@ def __init__(self, pin=None, active_high=True, initial_value=0, frequency=100):
         self._controller = None
         if not 0 <= initial_value <= 1:
             raise OutputDeviceBadValue("initial_value must be between 0 and 1")
-        super(PWMOutputDevice, self).__init__(pin, active_high)
+        if initial_value in (0, 1):
+            super(PWMOutputDevice, self).__init__(pin, active_high, initial_value=initial_value)
```

Comment:
> I vaguely recall trying it a while back and realizing it wasn't a 30 second job so I left it for the time being

Doing a quick grep in the sourcecode, it seems the only child-classes of OutputDevice are DigitalOutputDevice, PWMOutputDevice and SPISoftwareInterface.
So I guess one approach might be to make OutputDevice an abstract base class (see http://gpiozero.readthedocs.io/en/latest/api_output.html#base-classes) which isn't directly instantiable (which will mean tweaking a whole bunch of tests), and then move `output_with_state` to only be in the DigitalOutputDevice constructor?  I may have a play later this evening...
(Hmmm, I wonder if `_active_state` and `_inactive_state` could to be moved to only be in DigitalOutputDevice (and DigitalInputDevice) as well?)